utils/_context/virtual_machines.py

```diff
@@ -0,0 +1,155 @@
+import os
```

Comment:
Could you use the `__init__/configure` design pattern : 

* __init__ never fails and is always called, even if the class won't be used
* configure can (should) fails, and will be called if and only if the class will be used